src/uint/mul.rs

```diff
@@ -46,15 +46,24 @@ impl<const LIMBS: usize> Uint<LIMBS> {
     }
 
     /// Perform wrapping multiplication, discarding overflow.
-    // #[inline(always)]
     pub const fn wrapping_mul<const RHS_LIMBS: usize>(&self, rhs: &Uint<RHS_LIMBS>) -> Self {
         karatsuba::wrapping_mul_fixed::<LIMBS>(self.as_uint_ref(), rhs.as_uint_ref()).0
     }
 
     /// Perform saturating multiplication, returning `MAX` on overflow.
     pub const fn saturating_mul<const RHS_LIMBS: usize>(&self, rhs: &Uint<RHS_LIMBS>) -> Self {
-        let (res, overflow) = self.widening_mul(rhs);
-        Self::select(&res, &Self::MAX, overflow.is_nonzero())
+        self.checked_mul(rhs).unwrap_or(Uint::MAX)
+    }
+
+    /// Perform wrapping multiplication, checking that the result fits in the original [`Uint`] size.
+    pub const fn checked_mul<const RHS_LIMBS: usize>(
```

Comment:
FYI @tarcieri This method can clobber the trait method, returning a `ConstCtOption` instead of a `CtOption`. It's consistent with the `checked_square` method though, and provides `const` support. I don't think the method is widely used but in some cases this would produce an unneeded import warning for `CheckedMul`, in others it could produce an error if a `CtOption` was expected.